oasis-reference/src/main/kotlin/com/empowerops/volition/ref_oasis/ConnectionView.kt

```diff
@@ -0,0 +1,125 @@
+import kotlinx.coroutines.launch
+import tornadofx.*
+
+class ConnectionsView : View("My View") {
+    private lateinit var dataModel: DataModelService
+    private lateinit var control: OptimizerEndpoint
+    private val regList: ObservableList<String> = FXCollections.observableArrayList()
+
+    override val root = listview<String> {
+        vgrow = Priority.ALWAYS
+        //TODO make this good looking item
+        isEditable = false
+        cellFormat {
+            graphic = vbox {
+                hbox {
+                    label("Name: ")
+                    label(it)
+                }
+                hbox {
+                    label("Description: ")
+                    textflow {
+                        text(dataModel.simByName.getValue(it).description)
+                    }
+                }
+                hbox {
+                    label("Inputs: ")
+                    vbox {
+                        dataModel.simByName.getValue(it).inputs.forEach { input ->
+                            add(label("${input.name} [${input.lowerBound}, ${input.upperBound}] "))
+                        }
+                    }
+                }
+                hbox {
+                    label("Outputs: ")
+                    vbox {
+                        dataModel.simByName.getValue(it).outputs.forEach { output ->
+                            add(label(output.name))
+                        }
+                    }
+                }
+                hbox {
+                    spacing = 5.0
+                    button("refresh") {
+                        action {
+                            GlobalScope.launch {
+                                control.updateNode(it)
+                                dataModel.updateConfiguration(it)
+                            }
+                        }
+                    }
+                    button("delete") {
+                        action {
+                            GlobalScope.launch {
+                                dataModel.closeSim(it)
+                            }
+                        }
+                    }
+                    button("add setup") {
+                        action {
+                            GlobalScope.launch {
+                                dataModel.addConfiguration(it)
+                                dataModel.updateConfiguration(it)
+                            }
+                        }
+                    }
+                    button("remove setup") {
+                        action {
+                            GlobalScope.launch {
+                                dataModel.removeConfiguration(it)
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+
+    @Subscribe
+    fun whenNewNodeRegistered(event : PluginRegisteredEvent){
+        GlobalScope.launch(Dispatchers.JavaFx) {
+            regList.add(event.name)
+        }
+    }
```

Comment:
interesting, so in this configuration you have the view do the view-binding logic we traditionally leave for controllers.

What do you think about this? Given that our view so closely resembles the model and the domain events can be expressed very quickly in view terms, this is probably fine.

Is the same true of view events: can view events be expressed in domain terms just as simply?

Can we get away with MV, or maybe we're aiming for MVVM?

Reply:
Should we have a controller for  TornadoFx view? Yes, technically this is a view + controller. Since we haven't use any dependency injection framework there probably need a bit configuration(imagining this is fxml get controller via reflection) via `setData()` call that we have but we can totally have a controller doing all these.